src/work/WorkScheduler.cpp

```diff
@@ -45,14 +50,32 @@ WorkScheduler::scheduleOne(std::weak_ptr<WorkScheduler> weak)
     }
 
     self->mScheduled = true;
-    self->mApp.getClock().getIOContext().post([weak]() {
+    self->mTriggerTimer.expires_from_now(TRIGGER_PERIOD);
+    self->mTriggerTimer.async_wait([weak](asio::error_code) {
         auto innerSelf = weak.lock();
         if (!innerSelf)
         {
             return;
         }
+
+        try
+        {
+            // loop as to perform some meaningful amount of work
+            YieldTimer yt(innerSelf->mApp.getClock(),
+                          std::chrono::milliseconds(1));
```

Comment:
There are a couple of things here to note.

There is the throttling of how often we schedule this callback (done as to avoid doing too much "Work" related things per unit of time). This is driven by something like "apply bucket" that is very aggressive. The current throttle of `TRIGGER_PERIOD` (50ms) seems to be just fine for this as catchup performs pretty much the same than without throttling.

Then there is this loop: it's there to avoid bouncing off the main queue for trivial state changes that the `Work` tree needs to do (like when it cycles through parent/child `onRun`). 1ms is plenty of time for this kind of transitions (and it basically breaks as soon as it hits an "expensive" item such as anything that touches the database).

This was needed as we now throttle when we bounce off the main queue (in master, those bounces were only counted towards the main "batch size").

Reply:
Ok but the fact that 1-ms-CPU-per-50ms is "just fine" for bucket apply is a coincidence -- that a single step of bucket-apply happens to run a pretty coarse-grained step of activity -- and it wouldn't be true if we had a work step that was finer-grained.

I think at very least there ought to be a comment about this: if you break your work into units that themselves run for less-than-1ms you might well find your work takes a _very_ long time to run.

But .. there's something deeper I think.

All this playing with reschedule timers and the VirtualClock _event_ queue is making me very uncomfortable. Something's wrong about it. I think .. anything that's using such "self-reschedule" timers -- not using a timer to wait for a definite virtual-time _deadline_ of some sort, but just posting themselves back  to "run the next step" at some _frequency_ -- is doing so because they're trying to approximate a static schedule bound. That made a little bit of sense when we had no control over scheduling in ASIO's queue, but the whole point of this PR is that we're growing our own round-robin scheduler adjacent to ASIO's work queue, so we can control it. Wouldn't it make more sense to just make sure WorkScheduler gets a queue in the RR scheduler, and intermix its steps in RR fashion, and avoid (ab)using timers this way?

(We could also modify the RR scheduler to have quantums/budgets or priorities, there are a lot of RR variants)